### ISSUE TITLE
Auth: make redirect_uri use https if connected over https

### ISSUE THREAD

**Issue by inodb** [cBioPortal/cbioportal#6342]
Right now redirect_uri always uses http. This causes issues when using e.g. microsoft authentication. One needs to then specifically set `--proxy-base-url` to https e.g. https://github.com/knowledgesystems/knowledgesystems-k8s-deployment/blob/master/cbioportal/cbioportal_spring_boot.yaml#L120-L124

The logic should be:

- redirect to http when coming from http
- redirect to https when coming from https

If this is not possible then the default should prolly be https. Do note that we still have several instances using http, so we should have some way to override it (e.g. pdx/cbioportal.org/pdx)

**Comment by jjgao** [comment]
should we always use https? What's the benefit of supporting both http and https?

**Comment by inodb** [comment]
I think the logic should be:

- redirect to http when coming from http
- redirect to https when coming from https

**Comment by inodb** [comment]
@pvannierop and @oplantalech ran into the same issue with saml

**Comment by pvannierop** [comment]
Yes, we run our container this way now (configured via .env file):
```
/bin/sh -c "java ${JAVA_OPTS} -jar /webapp-runner.jar --proxy-base-url ${PROXY_BASE_URL} /cbioportal-webapp"
```

**Comment by inodb** [comment]
For reference. We use the same solution on our production machine on AWS: https://github.com/knowledgesystems/knowledgesystems-k8s-deployment/blob/master/cbioportal/cbioportal_spring_boot.yaml#L125-L126

**Comment by inodb** [comment]
Thanks for the pointer @mdorr - added a small note to that section of the docs for now

**Comment by j-hudecek** [comment]
If you're running cBioPortal in an old-school Tomcat process behind a reverse proxy you need to make sure `X-Forwarded-Proto` HTTP header is set by your reverse proxy and add the RemoveIpValve to your configuration (cbioportal-specific or `server.xml`)
https://tomcat.apache.org/tomcat-7.0-doc/api/org/apache/catalina/valves/RemoteIpValve.html
with at least 

     <Valve className="org.apache.catalina.valves.RemoteIpValve" 
          protocolHeader="x-forwarded-proto" />
